src/tuned_models.jl

```diff
@@ -238,26 +265,40 @@ _length(::Nothing) = 0
 # builds on an existing `history` until the length is `n` or the model
 # supply is exhausted (method shared by `fit` and `update`). Returns
 # the bigger history:
-function build(history, n, tuning, model::M,
-               state, verbosity, acceleration, resampling_machine) where M
+function build(history,
+               n,
+               tuning,
+               model,
+               state,
+               verbosity,
+               acceleration,
+               resampling_machine)
     j = _length(history)
     models_exhausted = false
     while j < n && !models_exhausted
-        _models = models!(tuning, model, history, state, verbosity)
-        models = _models === nothing ? M[] : collect(_models)
-        Δj = length(models)
+        metamodels = models!(tuning, model, history, state, verbosity)
+        if metamodels === nothing
```

Comment:
pretty sure here you could use the `_length` function that is defined above?

Reply:
Yep. Thanks.